include/oneapi/dpl/experimental/kt/internal/esimd_radix_sort_kernels.h

```diff
@@ -0,0 +1,722 @@
+                __dpl_esimd::__ns::simd<_GlobOffsetT, __bin_width> after_group_hist_sum = __prev_group_hist_sum + __thread_grf_hist_summary;
+                __dpl_esimd::__block_store<::std::uint32_t, __bin_width>(__p_this_group_hist + __local_tid * __bin_width,
+                                                       after_group_hist_sum | __hist_updated | __global_accumulated);
+
+                __dpl_esimd::__block_store_slm<::std::uint32_t, __bin_width>(
+                    __slm_bin_hist_global_incoming + __local_tid * __bin_width * sizeof(_GlobOffsetT), __prev_group_hist_sum);
+            }
+            __dpl_esimd::__ns::barrier();
+        }
+        // 3. Get total offsets for each work-item
+        auto __group_incoming = __dpl_esimd::__block_load_slm<_LocOffsetT, __bin_count>(__slm_bin_hist_group_incoming);
+        // 3.1. Get histogram accumullated from previous groups (together with the global one) and apply correction for keys already reordered in SLM
+        // TODO: rename the variable to represent its purpose better.
+        __global_fix = __dpl_esimd::__block_load_slm<_GlobOffsetT, __bin_count>(__slm_bin_hist_global_incoming) - __group_incoming;
+        // 3.2 Get historam with offsets for each work-item within its work-group.
+        if (__local_tid > 0)
+        {
+            __subgroup_offset = __group_incoming + __dpl_esimd::__block_load_slm<_LocOffsetT, __bin_count>((__local_tid - 1) * __hist_stride);
+        }
+        else
+        {
+            __subgroup_offset = __group_incoming;
+        }
+        __dpl_esimd::__ns::barrier();
+    }
+
+    template<typename _SimdPack>
+    void inline
+    __reorder_reg_to_slm(const _SimdPack& __pack, const _LocOffsetSimdT& __ranks,
+                         const _LocOffsetSimdT& __bins, const _LocHistT& __subgroup_offset,
+                         ::std::uint32_t __wg_size, ::std::uint32_t __thread_slm_offset) const
+    {
+        __slm_lookup<_LocOffsetT> __subgroup_lookup(__thread_slm_offset);
+        _LocOffsetSimdT __wg_offset =
+            __ranks + __subgroup_lookup.template __lookup<__data_per_work_item>(__subgroup_offset, __bins);
+        __dpl_esimd::__ns::barrier();
+
+        _GlobOffsetSimdT __wg_offset_keys = __wg_offset * sizeof(_KeyT);
+        __dpl_esimd::__vector_store<_KeyT, 1, __data_per_work_item>(__wg_offset_keys, __pack.__keys);
+        if constexpr (__has_values)
+        {
+            _GlobOffsetSimdT __wg_offset_vals = __wg_size * __data_per_work_item * sizeof(_KeyT) +
+                                                __wg_offset * sizeof(_ValT);
+            __dpl_esimd::__vector_store<_ValT, 1, __data_per_work_item>(__wg_offset_vals, __pack.__vals);
+        }
+        __dpl_esimd::__ns::barrier();
+    }
+
+    void inline
+    __reorder_slm_to_glob(const _GlobHistT& __global_fix, ::std::uint32_t __local_tid, ::std::uint32_t __wg_size) const
+    {
+        auto __pack = __make_simd_pack<__data_per_work_item, _KeyT, _ValT>();
```

Comment:
Is there a reason we call `__make_simd_pack` in the helper function rather than pass it in like is done in some of the other helpers?

Reply:
Fixed